library/ssl_msg.c

```diff
@@ -2048,7 +2048,7 @@ int mbedtls_ssl_fetch_input( mbedtls_ssl_context *ssl, size_t nb_want )
             if( ret < 0 )
                 return( ret );
 
-            if ( (size_t)ret > len || ( INT_MAX > SIZE_MAX && ret > SIZE_MAX ) )
+            if ( (size_t)ret > len || ( INT_MAX > SIZE_MAX && (size_t)ret > SIZE_MAX ) )
```

Comment:
I wonder why neither gcc nor clang complain here. Perhaps because we only ever run them on platforms where `INT_MAX > SIZE_MAX` is false and they optimize the `&&` before running the analysis on the right-hand size that would trigger the sign-compare warning.

Reply:
Looks like on e2k it's also false:
```
sakuya ~ # cat intmax.c ; gcc intmax.c -o intmax && ./intmax
#include<stdio.h>
#include<limits.h>
#include<stdint.h>
int main() { printf("%lu, %lu\n", INT_MAX, SIZE_MAX); return 0; }
2147483647, 18446744073709551615
sakuya ~ #
```
But I think warning's caused by lcc's language parser, which is probably run before any optimization stages; gcc may eliminate warnings for a reduced code in this case, but lcc probably doesn't.